mintscript/src/translate.rs

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common::chain::{
+    block::BlockRewardTransactable, signature::inputsig::InputWitness, tokens::TokenId,
+    AccountCommand, AccountOutPoint, AccountSpending, DelegationId, Destination, PoolId,
+    SignedTransaction, TxOutput, UtxoOutPoint,
+};
+use pos_accounting::PoSAccountingView;
+use tokens_accounting::TokensAccountingView;
+use utxo::Utxo;
+
+use crate::WitnessScript;
+
+/// An error that can happen during translation of an input to a script
+#[derive(thiserror::Error, Debug, PartialEq, Eq)]
+pub enum TranslationError {
+    #[error("Attempt to spend an unspendable output")]
+    Unspendable,
+
+    #[error("Illegal account spend")]
+    IllegalAccountSpend,
+
+    #[error("Illegal output spend")]
+    IllegalOutputSpend,
+
+    #[error(transparent)]
+    PoSAccounting(#[from] pos_accounting::Error),
+
+    #[error(transparent)]
+    TokensAccounting(#[from] tokens_accounting::Error),
+
+    #[error("Stake pool {0} does not exist")]
+    PoolNotFound(PoolId),
+
+    #[error("Delegation {0} does not exist")]
+    DelegationNotFound(DelegationId),
+
+    #[error("Token with id {0} does not exist")]
+    TokenNotFound(TokenId),
+}
+
+/// Contextual information about given input
+pub enum InputInfo<'a> {
```

Comment:
I would call this `InputStorageSourceType` or similar... I'm trying to avoid generic words like "info"

Reply:
Not sure, it's contextual information about the input, i.e. input paired up with some extra things needed for verification such as utxo. I agree that the name is not great but I don't see you suggestion improving it, it's not a `*SourceType` in any way as far as I can see.

Reply:
Can anyone suggest a better name?